app/lib/exchange-rate/money.ts

```diff
@@ -0,0 +1,458 @@
+import { Big } from 'big.js';
+
+export type NumberInput = number | string | Big;
+
+export type MoneyInput = {
+  /**
+   * Money amount
+   */
+  amount: NumberInput;
+  /**
+   * Currency for the provided amount of money
+   */
+  currency: string;
+  /**
+   * Unit of currency to use. For example for USD it can be 'dollar' or 'cent', for BTC 'bitcoin', 'satoshi' or
+   * 'millisatoshi', etc.
+   * If not provided the default/base unit is used (bitcoin for BTC, dollar for USD, etc.)
+   */
+  unit?: string;
+};
+
+type FormatOptions = {
+  locale?: string;
+  currency?: string;
+};
+
+type CurrencyUnit = {
+  name: string;
+  decimals: number;
+  symbol: string;
+  factor: Big;
+  format: (value: number, options?: FormatOptions) => string;
+};
+
+type CurrencyData = {
+  baseUnit: string;
+  units: CurrencyUnit[];
+};
+
+type BaseFormatOptions = FormatOptions & { decimals: number };
+
+function baseFormat(value: number, options: BaseFormatOptions) {
+  const { locale, decimals, currency } = options;
+  const formatOptions: Parameters<typeof Intl.NumberFormat>[1] = {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  };
+  if (currency) {
+    formatOptions.style = 'currency';
+    formatOptions.currency = currency;
+  }
+  return Intl.NumberFormat(locale, formatOptions).format(value);
+}
+
+const currencyDataMap: Record<string, CurrencyData> = {
+  USD: {
+    baseUnit: 'dollar',
+    units: [
```

Comment:
I am not sure if units is the right name for this. What do you think?

I am also not sure about unit names being `dollar`, `cent`, `bitcoin`, `satoshi` and `millisatoshi`. Maybe we should change it to `usd`, `cent` (or `ct`), `btc`, `sat` and `msat`?

Reply:
updated to usd, cent, btc, sat and msat